js/classes/entities/midground/enemies/slime.js

```diff
@@ -87,56 +75,66 @@ class Slime {
         return 3;
     };
 
-    // /** This will be the height of the Slime ON THE CANVAS. */
-    // static get SCALED_HEIGHT() {
-    //     return Slime.SCALE * Slime.HEIGHT;
-    // };
-
-    // /** This will be the width of the Slime ON THE CANVAS. */
-    // static get SCALED_WIDTH() {
-    //     return Slime.SCALE * Slime.WIDTH;
-    // };
-
     /** This will be the size of the Slime ON THE CANVAS. */
     static get SCALED_SIZE() {
         return Vector.multiply(Slime.SIZE, Slime.SCALE);
     }
 
+    /** The speed of a Slime in pixels/s. */
     static get SPEED() {
         return Slime.SCALE * 30;
     }
 
+    /** The maximum health of a Slime. */
+    static get MAX_HEALTH() {
+        return 20;
+    }
+
+    /** The distance (in pixels) that a Slime will stray from its initial position when roaming. */
+    static get PACE_DISTANCE() {
+        return Slime.SCALED_SIZE.x * 5;
+    }
+
+     /** The number of seconds between attacks. */
+     static get ATTACK_COOLDOWN() {
+        return 1;
+    };
+
+    /** The amount of damage a Slime deals during an attack. */
+    static get ATTACK_DAMAGE() {
+        return 5;
+    };
+
     /** The filepath to the spritesheet of the Slime. */
     static get SPRITESHEET() {
         // TODO: make Slime death sprite.
         return "./sprites/slimes.png";
     };
+
+    /**
+     * Perform any necessary operations when the Slime dies.
+     */
+    handleDeath() {
+        // if the enemy is now dead, remove it from the game
+        // TODO: replace this code with any death effects, state changes, etc.
+        this.removeFromWorld = true;
+    }
     
     /** Change what the Slime is doing and where it is. */
     update() {
-        // WHAT SHOULD HIS CONDITIONS BE?
+        this.base.update();
 
-        // He's gonna (wait 0.5s, move) x2, turn, repeat.
-        if (this.elapsedTime % 1.5 < 0.5) {
-            this.action = "idle";
-        } else {
-            this.action = "moving";
-        }
-        // His animation takes 1 second. So: 
-        this.hops = Math.floor(this.elapsedTime / 1.5);
-        this.facing = (Math.floor(this.hops / 2) % 2 === 0) ? "right" : "left";
-        this.elapsedTime += GAME.clockTick;
-
-        // HOW SHOULD WE MOVE HIM BASED ON HIS CONDITIONS?
-        const mult = (this.facing === "left") ? -1 : 1;
-        if (this.action === "moving") {
-            this.pos = Vector.add(this.pos, new Vector(mult * Slime.SPEED * GAME.clockTick, 0));
+        if (this.base.chadDistance() < Slime.SCALED_SIZE.x / 2 
```

Comment:
Should the cooldown be using Base and not Slime?

Reply:
This code makes use of the `chadDistance` method of `EnemyBase`, which finds the distance between the enemy and Chad adjusted for character height.